lms/models/lti_user.py

```diff
@@ -24,6 +25,24 @@ class LTIUser(NamedTuple):
     email: str = ""
     """The user's email address."""
 
+    validation_error: ValidationError = None
+    """Exception found while trying to construct a LTIUser from the request"""
+
+    def __bool__(self):
+        """
+        Override the bool behaviour of LTIUser.
+
+        In case when we want to return an "empty" LTIUser behave like false so checks like:
+
+            `if request.lti_user:`
+
+        produce the expected result.
+        """
+        # Behave as "falsy" if neither the user_id or application_instance are present
+        return bool(
+            self.user_id and self.application_instance_id and not self.validation_error
+        )
```

Comment:
Hmmmmm. The complex definition of falsey feels wrong to me. It seems like an `LTIUser` could have certain attributes (like an email) and still be falsey. Plus I think Pyramid already has a way of saying "no user" or "falsey user": `request.lti_user` is `None`